lib/events.js

```diff
@@ -426,9 +426,16 @@ function spliceOne(list, index) {
   list.pop();
 }
 
-function arrayClone(arr, i) {
-  var copy = new Array(i);
-  while (i--)
-    copy[i] = arr[i];
-  return copy;
-}
+// Optimized common usages (listeners < 50) and using slice for
+// large arrays so it scales much better
+function arrayClone(arr, length) {
+  if (length === 1) {
+    return [arr[0]];
+  } else if (length < 50) {
```

Comment:
I like to use elseif in those cases so the code looks more compacted in easy functions. Anyway in order to add a comment in the second case a simple if looks better.